### PR TITLE
edit fleet resnet train script

### PR DESCRIPTION
**Pull request by danleifeng**
in order to keep consistent with benchmark training script, some changes are added here.
- add benchmark_test param to control whether to print benchmark_logs.
- test only run in last epoch for saving time.
- dist_strategy is added for test_prog to speed up testing.
- lr_warmup is added to improve model test accuracy.

### CODE REVIEW

File: benchmark/collective/resnet/train_with_fleet.py [hunk 1]
```diff
@@ -308,15 +307,10 @@ def build_program(is_train, main_prog, startup_prog, args, dist_strategy=None):
...
-                    params_grads = optimizer.backward(avg_cost, startup_prog)
-                    master_params_grads = create_master_params_grads(
-                        params_grads, main_prog, startup_prog, args.scale_loss)
-                    optimizer.apply_gradients(master_params_grads)
-                    master_param_to_train_param(master_params_grads,
-                                                params_grads, main_prog)
-                else:
-                    dist_optimizer = fleet.distributed_optimizer(optimizer, strategy=dist_strategy)
-                    _, param_grads = dist_optimizer.minimize(avg_cost)
+		    dist_strategy.use_mixed_precision = True
```
Comment (gongweibao) [inline benchmark/collective/resnet/train_with_fleet.py]: 这个地方可以先保持不变。
等AMP稳定成熟一起改动。

File: benchmark/collective/resnet/train_with_fleet.py [hunk 2]
```diff
@@ -95,26 +93,27 @@ def set_paddle_flags(flags):
...
+        batch_denom = 256
+        start_lr = params["lr"]
+        base_lr = params["lr"] * global_batch_size / batch_denom
         lr = [base_lr * (0.1**i) for i in range(len(bd) + 1)]
-        optimizer = fluid.optimizer.Momentum(
-            learning_rate=fluid.layers.piecewise_decay(
-                boundaries=bd, values=lr),
-            momentum=momentum_rate,
-            regularization=fluid.regularizer.L2Decay(l2_decay))
+        lr_var = lr_warmup(fluid.layers.piecewise_decay(boundaries=bd, values=lr),\
```
Comment (gongweibao) [inline benchmark/collective/resnet/train_with_fleet.py]: 之前是有问题是吗？

File: benchmark/collective/resnet/train_with_fleet.py (hunk 1 above)
Comment (danleifeng) [reply benchmark/collective/resnet/train_with_fleet.py]: done.

File: benchmark/collective/resnet/train_with_fleet.py (hunk 2 above)
Comment (danleifeng) [reply benchmark/collective/resnet/train_with_fleet.py]: influence a little. tested by shenliang.

File: benchmark/collective/resnet/train_with_fleet.py [hunk 3]
```diff
@@ -47,12 +47,10 @@ def set_paddle_flags(flags):
 from paddle.fluid.contrib.mixed_precision.decorator import decorate
 import utils.reader_cv2 as reader
 from utils.utility import add_arguments, print_arguments, check_gpu
-from utils.learning_rate import cosine_decay_with_warmup
-from paddle.fluid.contrib.mixed_precision.fp16_lists import black_list, white_list, gray_list
```
Comment (gongweibao) [inline benchmark/collective/resnet/train_with_fleet.py]: 这几行貌似不能删除。

Comment (danleifeng) [reply benchmark/collective/resnet/train_with_fleet.py]: change into decorator.
